design/FY2022/NFP-Python-HVAC-Diagram.md

```diff
@@ -0,0 +1,85 @@
+# Python HVAC Diagram #
+
+Jason DeGraw, Piljae Im, and Niraj Kunwar
+
+Oak Ridge National Laboratory
+
+Original Date: November 18, 2021
+
+## Justification for New Feature ##
+
+The HVAC-Diagram program is a Fortran utility that can be used to generate
+an SVG representation of the HVAC system in a model based on the BND file
+generated by EnergyPlus. The program has limited flexibility and the SVG
+output is not correctly rendered by all SVG rendering libraries. The SVG format
+[1] is a large and complex format with many capabilities but is somewhat
+overpowered for this application.
+
+## Overview ##
+
+A Python replacement for the utility will be developed to produce output similar
+to the functionality of the original utility. While SVG output will be considered,
+prioirity will be placed on production of output that conveys the same information
+in a more reliable manner.
+
+## Approach ##
+
+The BND output from EnergyPlus will be used to generate several forms of output:
+
+* One or more DOT language [2] files. This format is used by the
+  open source Graphviz package to display and manipulate graphs. Other formats
+  will be considered depending on development team feedback.
+* An HTML-based representation, if possible in a single file. Modern browser
+  capabilities can be leveraged to produce semi-interactive representation of the
+  HVAC loops (e.g. simple controls to turn on/off node names or other ancilliary
+  information). Numerous visualization libraries are available (e.g. D3, p5.js) as
+  well as more traditional techniques (e.g. canvas, imagemaps).
```

Comment:
Letting the user be able to zoom into part of the diagram that is important to them is good and an advantage to SVG. Can this be done with canvas or imagemaps?

Reply:
Yes, it depends on how complex an approach is taken. If the Javascript path is taken, there're a lot of possibilities.